### PR TITLE
fix username keyerror (github kuma issue #6623)

### PR DESCRIPTION
**Pull request by escattone**
@peterbe This was a thought experiment for partly addressing #6623 (it doesn't address the failure to recover existing accounts, just the immediate server error) in a way that is more radical and cleans-up some code that has been bothering me for some time, but I'm not sure I like it. I'm submitting it as an example for discussion.

### CODE REVIEW

File: kuma/users/views.py [hunk 1]
```diff
@@ -599,31 +599,31 @@ def get_form(self, form_class=None):
...
-        # When no username is provided, default to the local-part of the email address
-        if form.initial.get("username", "") == "":
-            email = form.initial.get("email", "")
-            if isinstance(email, tuple):
-                email = email[0]
-            suggested_username = suggested_username_base = email.split("@")[0]
+        # We should only see GitHub/Google users.
+        assert self.sociallogin.account.provider in ("github", "google")
+
+        initial_username = form.initial.get("username")
```
Comment (peterbe) [inline kuma/users/views.py]: Down on line 624 there's that strange...
```python
if initial_username:
    ...
else:
    initial_username = ""
```
I think a more idiomatic syntax would be to use 
```suggestion
        initial_username = form.initial.get("username") or ""
```

File: kuma/users/views.py [hunk 2]
```diff
@@ -599,31 +599,31 @@ def get_form(self, form_class=None):
...
+                initial_username = f"{initial_username_base}{increment}"
+        else:
+            initial_username = ""
 
-        # For GitHub/Google users, see if we can find matching user by username
-        assert self.sociallogin.account.provider in ("github", "google")
-        try:
-            self.matching_user = User.objects.get(username=initial_username)
-            # deleting the initial username because we found a matching user
-            del form.initial["username"]
```
Comment (peterbe) [inline kuma/users/views.py]: With this, you can never get a KeyError on `form.initial['username']` any more. 👍🏼

**Comment by peterbe** [comment]
Start with this :) https://github.com/escattone/kuma/pull/4

File: kuma/users/views.py (hunk 1 above)
Comment (escattone) [reply kuma/users/views.py]: Yep, that's better @peterbe, thanks. I think instead but almost the same I'll use:
```python
initial_username = form.initial.get("username", "")
```

Comment (peterbe) [reply kuma/users/views.py]: But then `intial_username` might become `None` something else falsy. 
But storm in a teacup, I don't think it's a risk in the current context.

Comment (escattone) [reply kuma/users/views.py]: Ah, yes, I suppose the actual value of the `username` key could be `None` so yours is safer. Sorry, I'll adjust that in my clean-up PR.

**Pull request merged**